### PR TITLE
Add to blueprinting-tips

### PR DESCRIPTION
**Pull request by aledsage**

### CODE REVIEW

File: guide/yaml/blueprinting-tips.md [hunk 1]
```diff
@@ -12,6 +12,19 @@ The use of Java is reserved for those use-cases where the provisioning or config
 is very complicated.
 
 
+## Be Familiar with Brooklyn
+
+Be familiar with the stock entities available in Brooklyn/AMP. For example, prove you understand
```
Comment (drigodwin) [inline guide/yaml/blueprinting-tips.md]: I would remove AMP from here

File: guide/yaml/blueprinting-tips.md [hunk 2]
```diff
@@ -22,28 +35,30 @@ of artifacts (e.g. RPMs, zip files, etc).
 
 Options for speeding up provisioning include those below.
 
-### Deploying to the "localhost" location
+#### Deploying to Bring Your Own Nodes (BYON)
 
-This is fast and simple, but has some obvious downsides:
+A BYON location can be defined, which avoids the time required to provision VMs. This is fast,
```
Comment (drigodwin) [inline guide/yaml/blueprinting-tips.md]: Perhaps link out to the BYON section

File: guide/yaml/blueprinting-tips.md [hunk 3]
```diff
@@ -22,28 +35,30 @@ of artifacts (e.g. RPMs, zip files, etc).
...
+but has the downside that artifacts installed during a previous run can interfere with subsequent 
+runs.
 
-* Artifacts are installed directly on your desktop/server.
+A variant of this is to use Vagrant (e.g. with VirtualBox) to create VMs on your local machine,
```
Comment (drigodwin) [inline guide/yaml/blueprinting-tips.md]: Link out to [the distributed vagrant config](https://github.com/apache/brooklyn-dist/tree/master/vagrant)?

File: guide/yaml/blueprinting-tips.md [hunk 4]
```diff
@@ -54,26 +69,64 @@ container (i.e. no install artifacts from previous runs), while taking advantage
 of starting containers.
```
Comment (drigodwin) [inline guide/yaml/blueprinting-tips.md]: Link out to clocker.io?

File: guide/yaml/blueprinting-tips.md [hunk 5]
```diff
@@ -54,26 +69,64 @@ container (i.e. no install artifacts from previous runs), while taking advantage
...
+mega-blueprint. For example, ensure each component is added to the catalog independently, along 
+with a blueprint for the composite app.
+
+Experiment with lots of small blueprints to test independent areas before combining them into the 
+real thing.
+
+
+## Writing Entity Tests
+
+Use the test framework to write test cases. This will make automated (regression) testing easier,
```
Comment (drigodwin) [inline guide/yaml/blueprinting-tips.md]: Link out to the YAML testing section

File: guide/yaml/blueprinting-tips.md [hunk 6]
```diff
@@ -98,8 +151,32 @@ If writing a custom integration, the following recommendations may be useful:
...
+* Test (and regression test) on each target cloud.
+
+* Be aware that images on different clouds can be very different. For example, two CentOS 6.6 VMs 
+  might have different pre-installed libraries, different default iptables or SE Linux settings,
+  different repos, different sudo configuration, etc.
+
+* Different clouds handle private and public IPs differently. One must be careful about which 
+  address to advertise to for use by other entities.
+
+* VMs on some clouds will not have a well-configured hostname (e.g. `ping $(hostname)` can fail).
```
Comment (drigodwin) [inline guide/yaml/blueprinting-tips.md]: VMs on some clouds will not -> VMs on some clouds may not

**Comment by sjcorbett** [comment]
@aledsage Looks good. Duncan's comments worth addressing.

File: guide/yaml/blueprinting-tips.md (hunk 4 above)
Comment (aledsage) [inline guide/yaml/blueprinting-tips.md]: There is a link to `[Clocker project](http://www.clocker.io)` a few lines higher up.

**Comment by aledsage** [comment]
Thanks @drigodwin @sjcorbett - I've incorporated those changes. Merging now.

File: guide/yaml/blueprinting-tips.md (hunk 4 above)
Comment (drigodwin) [inline guide/yaml/blueprinting-tips.md]: Oh yes! sorry - it had been squashed and I missed it

**Pull request merged**